gusto/advection.py

```diff
@@ -185,10 +187,17 @@ def solve_stage(self, x_in, stage):
     @embedded_dg
     def apply(self, x_in, x_out):
 
+        if self.limiter is not None:
+            self.limiter.apply(x_in)
         self.q1.assign(x_in)
         for i in range(3):
             self.solve_stage(x_in, i)
+            if i < 2:
```

Comment:
I think this would be tidier if we amended the `solve_stage` method. For stage==2 you could assign to q1 as in the other stages and then do `x_out.assign(self.q1)` on line 198. You could either put the application of the limiter in this loop (without the need for the `if i < 2`) or put it in the `solve_stage` method, whichever you think is nicest.